packages/sdk-middleware-auth/src/base-auth-flow.js

```diff
@@ -0,0 +1,154 @@
+/* @flow */
+import type {
+  MiddlewareRequest,
+  MiddlewareResponse,
+  Next,
+  Task,
+} from 'types/sdk'
+
+/* global fetch */
+import 'isomorphic-fetch'
+
+type RequestState = boolean
+type TokenStore = {
+  token: string;
+  expirationTime: number;
+}
+type AuthMiddlewareBaseOptions = {
```

Comment:
Would it not be better to add the type-declarations in the `types` folder and import?